CMakeLists.txt

```diff
@@ -16,6 +21,7 @@ SET(CMAKE_BUILD_TYPE ${TARGET} CACHE STRING "Choose the target of build: Debug R
 SET(CRYPTO ${CRYPTO} CACHE STRING "Choose the crypto of build: mbedtls openssl" FORCE)
 SET(GCOV ${GCOV} CACHE STRING "Choose the target of Gcov: ON  OFF, and default is OFF" FORCE)
 SET(STACK_USAGE ${STACK_USAGE} CACHE STRING "Choose the target of STACK_USAGE: ON  OFF, and default is OFF" FORCE)
+SET(BUILD_SHARED_LIB ${BUILD_SHARED_LIB} CACHE STRING "Choose if shared library libspdm.co should be built: ON OFF, and default is OFF" FORCE)
```

Comment:
Good point. Windows shared lib is DLL.

I suggest we remove the `libspdm.so` term to avoid confusing. Just saying `shared library` is enough.

BTW: Does this PR support shared lib in windows?

Reply:
Hi, the change was built and tested on linux only. Pkg-config support is only available on linux also, as far as I know. I do not have the know-how to do the integration on Windows and I also don't have access to a windows machine and windows integration code to verify it works.
Are you guys ok with making this PR to enable only Linux shared library and pkg-config support? I can rename BUILD_SHARED_LIB to BUILD_LINUX_SHARED_LIB.

Reply:
I am OK to limit the support to Linux.

We just need better documentation to tell people.

Reply:
Hi, I will update the documentation as well in the next commit.